static/app/views/performance/newTraceDetails/index.tsx

```diff
@@ -126,21 +130,80 @@ export function TraceView() {
   );
 }
 
+type TraceWarningsProps = {
```

Comment:
I would split this into a TraceWarning component and move the util getTraceType function there as well.